Sources/Redis/Client/RedisClient+Pipeline.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+public final class RedisPipeline {
+    /// The commands in this pipeline
+    var commands = [RedisData]()
+    
+    /// Adds a command to the pipeline
+    public func command(_ command: String, args: [RedisData] = []) {
+        let command = RedisData.array([RedisData(bulk: command)] + args)
+        self.commands.append(command)
+    }
+    
+    /// Adds a command to the pipeline
+    public func command(_ command: String, args: RedisData...) {
+        self.command(command, args: args)
+    }
+}
+
+extension RedisClient {
+    /// Executes a pipelined transaction
+    public func pipeline(closure: (RedisPipeline)->()) -> Future<[RedisData]> {
+        // Fill the pipeline with commnads
+        let pipeline = RedisPipeline()
+        closure(pipeline)
+        
+        // Returns a empty fulfilled future if the transaction was empty
+        if pipeline.commands.count == 0 {
+            return eventLoop.newSucceededFuture(result: [RedisData]())
+        }
+        
+        // Send the commands as a pipeline
+        return send(pipeline: pipeline.commands)
+    }
+    
+    /// Executes a multi/exec transaction
+    public func multi(closure: (RedisPipeline)->()) -> Future<[RedisData]> {
+        // Fill the pipeline with commnads
+        let pipeline = RedisPipeline()
+        closure(pipeline)
+        
+        // Returns a empty fulfilled future if the transaction was empty
+        if pipeline.commands.count == 0 {
+            return eventLoop.newSucceededFuture(result: [RedisData]())
+        }
+        
+        // Create the multi-exec list of commands
+        let multiExec = [RedisData.array([.bulkString("MULTI")])] +
```

Comment:
Could we add `MULTI` support like all other command helpers?

Reply:
You could, then you would also just make a custom EXEC one as well and the developer would just call them all consecutively (ignoring the currentSend error).  To me, a group of transactions that are intended to be sent together are cleaner formed using a block/closure vs. a sequence of generic commands, but we clearly disagree on the API above so we will probably disagree on this one as well :).

Also typically it is ideal to build a "MULTI/EXEC" on top of a "pipeline" because the MULTI/EXEC is intended to execute as one continuous command on the server, meaning you should also submit it as 1 pipelined transaction which will make it only do 1 round trip to the server vs. once for every command.